### PR TITLE
[#493] improvement: replace `putIfAbsent` with `computeIfAbsent` to avoid performance loss in some critical paths

### PR DESCRIPTION
**Pull request by cchung100m**
Hi @zuston 

I have updated some critical paths with the replacement, however, there might be some missing pieces, any suggestions would be appreciated.

### What changes were proposed in this pull request?

Replace `putIfAbsent` with `computeIfAbsent` in some critical paths

### Why are the changes needed?

Fix: #493 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

current UT

### CODE REVIEW

File: client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java [hunk 1]
```diff
@@ -241,7 +241,7 @@ protected ShuffleBlockInfo createShuffleBlock(int partitionId, WriterBuffer wb)
 
   // it's run in single thread, and is not thread safe
   private int getNextSeqNo(int partitionId) {
-    partitionToSeqNo.putIfAbsent(partitionId, 0);
```
Comment (zuston) [inline client-spark/common/src/main/java/org/apache/spark/shuffle/writer/WriteBufferManager.java:244]: This is unnecessary.

File: client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java [hunk 2]
```diff
@@ -544,7 +544,7 @@ public void reportShuffleResult(
         groupedPartitions.get(ssi).add(partitionIdx);
       }
       if (CollectionUtils.isNotEmpty(partitionToBlockIds.get(partitionIdx))) {
-        partitionReportTracker.putIfAbsent(partitionIdx, 0);
+        partitionReportTracker.computeIfAbsent(partitionIdx, key -> 0);
```
Comment (zuston) [inline client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java:547]: ditto

File: client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java [hunk 3]
```diff
@@ -630,7 +630,7 @@ public Roaring64NavigableMap getShuffleResultForMultiPart(String clientType,
       ShuffleServerInfo shuffleServerInfo = entry.getKey();
       Set<Integer> requestPartitions = Sets.newHashSet();
       for (Integer partitionId : entry.getValue()) {
-        partitionReadSuccess.putIfAbsent(partitionId, 0);
+        partitionReadSuccess.computeIfAbsent(partitionId, key -> 0);
```
Comment (zuston) [inline client/src/main/java/org/apache/uniffle/client/impl/ShuffleWriteClientImpl.java:633]: ditto

File: common/src/main/java/org/apache/uniffle/common/util/RssUtils.java [hunk 4]
```diff
@@ -312,7 +312,7 @@ public static Map<Integer, Roaring64NavigableMap> generatePartitionToBitmap(
       Roaring64NavigableMap shuffleBitmap, int startPartition, int endPartition) {
     Map<Integer, Roaring64NavigableMap> result = Maps.newHashMap();
     for (int partitionId = startPartition; partitionId < endPartition; partitionId++) {
-      result.putIfAbsent(partitionId, Roaring64NavigableMap.bitmapOf());
+      result.computeIfAbsent(partitionId, key -> Roaring64NavigableMap.bitmapOf());
```
Comment (zuston) [inline common/src/main/java/org/apache/uniffle/common/util/RssUtils.java:315]: Nice! 👍

**Review by zuston — APPROVED** [review]
LGTM. Thanks for your contribution @cchung100m 🎉

**Comment by zuston** [comment]
Do you want join our WeChat group? If you are willing to do more contribution or have any good idea, feel free to propose.

IMG_9188

**Comment by cchung100m** [comment]
Hi @zuston 

Thanks for the invitation, I have joined the group.

**Pull request merged**